lib/url.go

```diff
@@ -0,0 +1,73 @@
+package lib
+
+// [RFC 1738](https://www.ietf.org/rfc/rfc1738.txt)
+// §2.2
+func shouldEscape(c byte) bool {
+	// alphanum
+	if 'A' <= c && c <= 'Z' || 'a' <= c && c <= 'z' || '0' <= c && c <= '9' {
+		return false
+	}
+
+	switch c {
+	case '$', '-', '_', '.', '+', '!', '*', '\'', '(', ')', ',': // Special characters
+		return false
+
+	case '/', '?', ':', '@', '=', '&': // Reserved characters
+		return c == '?'
+	}
+	// Everything else must be escaped.
+	return true
+}
+
+// PathEscapeRFC1738 escapes the string so it can be safely placed
+// inside a URL path segment according to RFC1738.
+// Based on golang native implementation of `url.PathEscape`
+// https://golang.org/src/net/url/url.go?s=7976:8008#L276
+func PathEscapeRFC1738(s string) string {
```

Comment:
Can you unexport this so this library suddenly doesn't become **the** PathEscapeRFC1738 implementation to use.

Reply:
Fixed